plugin.video.edltv/default.py

```diff
@@ -0,0 +1,317 @@
+        xbmcplugin.endOfDirectory(_handle)
+    elif len(category) == 7 :
+        set_period(category)
+    else :
+        lessons = get_data(category,None)
+        for data in lessons:
+            list_item = xbmcgui.ListItem(label=data[1])
+            url = get_url(action='listing', category=data[0], hasfolder='yes')
+            is_folder = True
+            xbmcplugin.addDirectoryItem(_handle, url, list_item, is_folder)
+        xbmcplugin.endOfDirectory(_handle)
+
+def list_videos(category) :
+    cloud_path = get_socket_data ()
+    if cloud_path :
+        cloud_path = cloud_path + '/Public/edltv/eDLTVA/htdocs/'
+        host_path = "smb://"
+    lessons = get_data(category,None)
+    if cloud_path :
+        for data in lessons :
+            list_item = xbmcgui.ListItem(label=data[1])
+            list_item.setInfo('video', {'title': data[1]})
+            list_item.setProperty('IsPlayable', 'true')
+            url = get_url(action='play', video=host_path+cloud_path+data[4])
+            is_folder = False
+            xbmcplugin.addDirectoryItem(_handle, url, list_item, is_folder)
+        xbmcplugin.endOfDirectory(_handle)
+    elif check_internet_connected() :
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok('eDLTV', 'ขั้นตอนนี้อาจมีค่าใช้บริการเมื่อต้องการเล่นวิดีโอ เนื่องจากมีการเชื่อมต่ออินเทอร์เน็ต')
+        for data in lessons :
+            list_item = xbmcgui.ListItem(label=data[1])
+            list_item.setInfo('video', {'title': data[1]})
+            list_item.setProperty('IsPlayable', 'true')
+            url = get_url(action='play', video=data[7])
+            is_folder = False
+            xbmcplugin.addDirectoryItem(_handle, url, list_item, is_folder)
+        xbmcplugin.endOfDirectory(_handle)
+    else :
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok('eDLTV', 'ไม่สามารถติดต่อกับเซิฟเวอร์ได้เนื่องจากไม่ได้เชื่อมต่อเครือข่ายหรือไม่มีการเชื่อมต่ออินเทอร์เน็ต')
+
+def play_video(path):
+    play_item = xbmcgui.ListItem(path=path)
+    xbmcplugin.setResolvedUrl(_handle, True, listitem=play_item)
+
+def set_search_list (l) :
+    cloud_path = get_socket_data ()
+    if cloud_path :
+        cloud_path = cloud_path + '/Public/edltv/eDLTVA/htdocs/'
+        host_path = "smb://"
+    
+    if cloud_path :
+        for data in l :
+            list_item = xbmcgui.ListItem(label=data[1])
+            list_item.setInfo('video', {'title': data[1]})
+            list_item.setProperty('IsPlayable', 'true')
+            url = get_url(action='play', video=host_path+cloud_path+data[3])
+            is_folder = False
+            xbmcplugin.addDirectoryItem(_handle, url, list_item, is_folder)
+        xbmcplugin.endOfDirectory(_handle)
+    elif check_internet_connected() :
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok('eDLTV', 'ขั้นตอนนี้อาจมีค่าใช้บริการเมื่อต้องการเล่นวิดีโอ เนื่องจากมีการเชื่อมต่ออินเทอร์เน็ต')
+        for data in l :
```

Comment:
it is usually a good practice to use `xbmcplugin.addDirectoryItems` instead of `xbmcplugin.addDirectoryItem` if you are adding multiple items. Just add them to a list and call addDirectoryItems afterwards by passing it the list